File: examples/typescript-express-pep/policy-test-ci.ts
import { createAccessKitClient } from "../../packages/typescript-client/src/index.js";

const apiKey = process.env.ACCESS_KIT_API_KEY;
const baseUrl = process.env.ACCESS_KIT_BASE_URL ?? "http://127.0.0.1:3000";
const policyId = process.argv[2];

if (!apiKey) {
  throw new Error("ACCESS_KIT_API_KEY is required.");
}

if (!policyId) {
  throw new Error("Usage: pnpm tsx examples/typescript-express-pep/policy-test-ci.ts <policy-id>");
}

const client = createAccessKitClient({ apiKey, baseUrl });
const result = await client.testPolicy(policyId, { correlationId: "corr:policy-test-ci" });
const failingChecks = result.checks.filter((check) => check.status === "fail");

if (!result.valid || failingChecks.length > 0) {
  throw new Error(`Policy ${policyId} failed ${failingChecks.length} policy-test checks.`);
}

console.log(`PASS ${policyId} policy-test checks.`);

File: examples/python-fastapi-pep/policy_test_ci.py
"""Run Access Kit policy tests from Python CI jobs."""

import os
import sys

from access_kit_pep import AccessKitClient


def main() -> int:
    api_key = os.environ.get("ACCESS_KIT_API_KEY", "")
    base_url = os.environ.get("ACCESS_KIT_BASE_URL", "http://127.0.0.1:3000")
    policy_id = sys.argv[1] if len(sys.argv) > 1 else ""

    if not api_key:
        raise SystemExit("ACCESS_KIT_API_KEY is required.")

    if not policy_id:
        raise SystemExit("Usage: python3 examples/python-fastapi-pep/policy_test_ci.py <policy-id>")

    client = AccessKitClient(base_url=base_url, api_key=api_key)
    result = client.test_policy(policy_id, correlation_id="corr:python-policy-test-ci")
    checks = result.get("checks", [])
    failing_checks = [
        check for check in checks
        if isinstance(check, dict) and check.get("status") == "fail"
    ]

    if not result.get("valid") or failing_checks:
        raise SystemExit("Policy %s failed %s policy-test checks." % (policy_id, len(failing_checks)))

    print("PASS %s policy-test checks." % policy_id)
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
